### PR TITLE
Added libtorrent and rtorrent formulas

### PR DESCRIPTION
**Pull request by zawapete**
libtorrent version 0.13.7
rtorrent version 0.9.7

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by zawapete** [comment]
please, not that when building through formula rtorrent suffers random crash
```
Caught Segmentation fault, dumping stack:
0   rtorrent                            0x000000010200b14f _Z8do_panici + 460
1   libsystem_platform.dylib            0x00007fffc959bb3a _sigtramp + 26
2   ???                                 0x000000010692e551 0x0 + 4405257553
3   rtorrent                            0x000000010204bc48 _ZN12ThreadWorkerC1Ev + 14
4   rtorrent                            0x0000000102009b39 main + 1086
5   libdyld.dylib                       0x00007fffc938c235 start + 1
6   ???                                 0x0000000000000001 0x0 + 1
Abort trap: 6```

anyway if I build rtorrent from the same source linking to the libtorrent build from the formula the binary is "stable"

```
CC=clang CXX=clang++ CXXFLAGS="-Wno-deprecated-declarations -O3 -std=c++11 -stdlib=libc++" LDFLAGS="-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk" ./configure --with-xmlrpc-c
make
make install
```

I'd like to investigate more after having some hints on what can be the issue, or in case push only libtorrent formula

**Comment by zawapete** [comment]
I'm fixing the jenkins build for sierra (the environment I'm working on), but I cannot test on high sierra e mojave: it seems an error related to cmath

what about the optional dependencies? `xmlrpc-c` is needed only if build `--with-xmlrpc-c`

**Comment by commitay** [comment]
> what about the optional dependencies?

If it's useful/popular make it the default, otherwise it should be removed.

**Comment by fxcoudert** [comment]
This cannot be named `libtorrent`. There are several projects named “libtorrent”, and this one is clearly not the most popular. A Google search for “libtorrent” brings up almost exclusively https://www.libtorrent.org which has a Wikipedia page and corresponds to this github repo: https://github.com/arvidn/libtorrent (with 1.5k stars). It is already in Homebrew core under the name `libtorrent-rasterbar`

Even scrolling down the Google results, finding this repo https://github.com/rakshasa/libtorrent is not easy. Two possibilities:
- if you think it has value in itself, make it a formula with a more specific name, e.g. `libtorrent-rtorrent`
- if it is only used in `rtorrent`, make it a resource inside the `rtorrent` formula

**Comment by commitay** [comment]
@fxcoudert This PR is re-adding deleted formulae with new versions. https://github.com/Homebrew/homebrew-core/pull/369

**Comment by zawapete** [comment]
the last error left in sierra build is on rtorrent install:
```
ld: file not found: /usr/lib/system/libsystem_darwin.dylib for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
I had this problem locally and I had to run again `xcode-select`, is it some floating issue on jenkins?

**Comment by commitay** [comment]
```diff
diff --git a/Formula/rtorrent.rb b/Formula/rtorrent.rb
index 3a1e5989bf..5d53ed1e26 100644
--- a/Formula/rtorrent.rb
+++ b/Formula/rtorrent.rb
@@ -13,6 +13,9 @@ class Rtorrent < Formula
   depends_on "xmlrpc-c"
 
   def install
+    # Fix file not found errors for /usr/lib/system/libsystem_darwin.dylib
+    ENV["SDKROOT"] = MacOS.sdk_path if MacOS.version == :sierra 
+
     ENV.append "CXXFLAGS", "-Wno-deprecated-declarations -O3 -std=c++11 -stdlib=libc++"
     ENV.append "LDFLAGS", "-isysroot #{MacOS.sdk_path}"
 
```

**Comment by zawapete** [comment]
thanks @commitay , I was juggling with that but didn't find the right solution

sierra build is fixed, as for the high sierra and mojave ones I'm investigating

**Comment by fxcoudert** [comment]
@commitay more than two years later, I think we can consider them “new formulas”, and apply those standards. I am concerned about adding a `libtorrent` formula which is not what 99% of our users will expect to be libtorrent.

**Comment by zawapete** [comment]
@fxcoudert I can easily rename to `libtorrent-rakshasa`

File: Formula/libtorrent-rakshasa.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  sha256 "c738f60f4d7b6879cd2745fb4310bf24c9287219c1fd619706a9d5499ca7ecc1"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+  depends_on "cppunit"
+  depends_on "openssl"
+
+  conflicts_with "libtorrent-rasterbar"
```
Comment (commitay) [inline Formula/libtorrent-rakshasa.rb]: Please also add the conflict to `libtorrent-rasterbar`.

It also needs an explanation, e.g:
https://github.com/Homebrew/homebrew-core/blob/a7bd0068dde3dfdd465d14ee612d709a7bc9d8e4/Formula/sonarqube.rb#L11

**Comment by zawapete** [comment]
I really don't understand how to fix the wrong cmath import on high sierra and mojave :(

**Comment by zawapete** [comment]
can you confirm that homebrew doesn't use `/usr/bin/clang++` in the build process?
I replicated the whole build process done by `brew install --build-from-source` when building directly from source and still the rtorrent binary built by brew crash randomly while mine not. the only difference left seems to be the actual compiler used

**Comment by zawapete** [comment]
I've found something about the build issue on `high sierra` and `mojave`
I've built form source on the first without problem, but by formula still fails.
I've narrowed down the difference to this:
```
Brew:
/tmp/libtorrent-rakshasa-20181014-14502-z3dyee/libtorrent-0.13.7//src/torrent/.deps/object_stream.Tpo:  /Library/Developer/CommandLineTools/usr/include/c++/v1/cmath \
/tmp/libtorrent-rakshasa-20181014-14502-z3dyee/libtorrent-0.13.7//src/torrent/.deps/object_stream.Tpo:  /usr/include/math.h ../../rak/algorithm.h ../../rak/string_manip.h \
/tmp/libtorrent-rakshasa-20181014-14502-z3dyee/libtorrent-0.13.7//src/torrent/.deps/object_stream.Tpo:/Library/Developer/CommandLineTools/usr/include/c++/v1/cmath:
/tmp/libtorrent-rakshasa-20181014-14502-z3dyee/libtorrent-0.13.7//src/torrent/.deps/object_stream.Tpo:/usr/include/math.h:

Build from sources:
./src/torrent/.deps/object_stream.Plo:  /Library/Developer/CommandLineTools/usr/include/c++/v1/cmath \
./src/torrent/.deps/object_stream.Plo:  /Library/Developer/CommandLineTools/usr/include/c++/v1/math.h \
./src/torrent/.deps/object_stream.Plo:  /usr/include/math.h ../../rak/algorithm.h ../../rak/string_manip.h \
./src/torrent/.deps/object_stream.Plo:/Library/Developer/CommandLineTools/usr/include/c++/v1/cmath:
./src/torrent/.deps/object_stream.Plo:/Library/Developer/CommandLineTools/usr/include/c++/v1/math.h:
./src/torrent/.deps/object_stream.Plo:/usr/include/math.h:
```

as you can see when building from brew `/Library/Developer/CommandLineTools/usr/include/c++/v1/math.h` is missing form the dependecy artifacts

I've tried to add `-I/Library/Developer/CommandLineTools/usr/include/c++/v1/math.h` with no luck

any hint?
I've seen it was a problem that other formula had before but the solutions I found related to that don't solve the issue with my formulas

**Comment by ericbn** [comment]
@zawapete, did you check this: https://gist.github.com/pyed/197f6a9348212e92c211ad3d3cff710a ?

**Comment by zawapete** [comment]
@ericbn that's not relevant anymore with ver 0.9.7
I'm able to compile without problem with no need for those flags, it only fails when building with brew

**Comment by zawapete** [comment]
any maintainer can give an hint about failing for `math.h`?

**Pull request closed without merging**